File: jimmer-ddd-java-sample/admin/src/main/java/org/morecup/jimmerddd/java/sample/admin/base/domaininterfaceimpl/SnowflakeManager.java
package org.morecup.jimmerddd.java.sample.admin.base.domaininterfaceimpl;

import cn.hutool.core.lang.Snowflake;
import org.morecup.jimmerddd.java.sample.domain.base.domaininterface.ISnowflakeManager;
import org.springframework.stereotype.Component;

@Component
public class SnowflakeManager implements ISnowflakeManager {
    private final Snowflake snowflake = new Snowflake(1, 1);

    @Override
    public long nextId() {
        return snowflake.nextId();
    }
}

File: jimmer-ddd-kotlin-sample/admin/src/main/kotlin/org/morecup/jimmerddd/kotlin/sample/admin/base/domaininterfaceimpl/SnowflakeManager.kt
package org.morecup.jimmerddd.kotlin.sample.admin.base.domaininterfaceimpl

import cn.hutool.core.lang.Snowflake
import org.morecup.jimmerddd.kotlin.sample.domain.base.domaininterface.ISnowflakeManager

object SnowflakeManager:ISnowflakeManager {
    private val snowflake = Snowflake(1, 1)
    override fun nextId(): Long {
        return snowflake.nextId()
    }
}